source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -340,10 +340,9 @@ namespace aspect
         const double unit_offset = 0.5;
         std::vector<double> property_minimums(n_particle_properties, std::numeric_limits<double>::max());
         std::vector<double> property_maximums(n_particle_properties, std::numeric_limits<double>::lowest());
-        for (typename ParticleHandler<dim>::particle_iterator particle = particle_range.begin();
-             particle != particle_range.end(); ++particle, ++particle_index)
+        for (const auto &particle : particle_range)
           {
-            const ArrayView<double> particle_property_value = particle->get_properties();
+            const ArrayView<double> particle_property_value = particle.get_properties();
```

Comment:
You accidentally improved the code above, but that means you also need to change this line to:
```suggestion
            const ArrayView<const double> particle_property_value = particle.get_properties();
```

Find me if you want to know more about why.